### PR TITLE
And on the seventh week he said "Let there be light"; and there was light (torch, lamp and treasure model)

### LINKED ISSUE #290: Dynamic lighting model - Lamp and torch model
Implement models for Lamps and torches.

This is a subtask of #289 - Dynamic lighting model

### PR DESCRIPTION
**Pull request by WytzeElhorst**
Adds the torch class (includes lamp), a test for this class and a model for the treasure.

This closes #290

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/model/entities/EntityType.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 	
 	TREASURE("Treasure", 			0,  Treasure::loadEntity),
 	WALLFRAME("WallFrame", 			0,  WallFrame::loadEntity),
+	TORCH("TORCH", 					0,  Torch::loadEntity),
```
Comment (WytzeElhorst) [inline src/main/java/nl/tudelft/contextproject/model/entities/EntityType.java]: I dont know why it does this, it looks good in eclipse

File: src/main/java/nl/tudelft/contextproject/model/entities/Torch.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+	@Override
+	public void move(float x, float y, float z) {
+		sp.move(x, y, z);
+		fire.move(x, y, z);
+		if (rb == null) getPhysicsObject();
+
+		rb.setPhysicsLocation(rb.getPhysicsLocation().add(x, y, z));
+	}
+	/**
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/Torch.java:97]: There should be an ampty line before this line

File: src/main/java/nl/tudelft/contextproject/model/entities/Torch.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	 * 		the position of the torch
+	 * @param data
+	 * 		the data of the torch
+	 * @return
+	 * 		the torch represented by the given data
+	 * @throws IllegalArgumentException
+	 * 		if the given data array is of incorrect length
+	 */
+	public static Torch loadEntity(Vector3f position, String[] data) {
+		if (data.length != 5) throw new IllegalArgumentException("Invalid data length for loading key! Expected \"<X> <Y> <Z> Torch <Type>\".");
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/Torch.java]: key -> torch

**Comment by Geweldig** [comment]
Other than my two extremely minor complaints I think this is good to merge!

**Comment by Diocruel** [comment]
Looks awesome. Code looks clean too. Merging.

File: src/main/java/nl/tudelft/contextproject/model/entities/Torch.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+	 * 		the data of the torch
+	 * @return
+	 * 		the torch represented by the given data
+	 * @throws IllegalArgumentException
+	 * 		if the given data array is of incorrect length
+	 */
+	public static Torch loadEntity(Vector3f position, String[] data) {
+		if (data.length != 5) throw new IllegalArgumentException("Invalid data length for loading torch! Expected \"<X> <Y> <Z> Torch <Type>\".");
+		Torch torch = null;
+		if (data[4] == "true") {
```
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/model/entities/Torch.java:166]: String comparison with ==

**Pull request merged**